### DISCUSSION TITLE
Migrating from Jaeger client to OpenTelemetry SDK

### DISCUSSION THREAD

**Discussion by Mulkave** [Vinelab/tracing-laravel#43] [Ideas]
Hey @KinaneD @adiachenko,

I stumbled upon [this article](https://medium.com/jaegertracing/migrating-from-jaeger-client-to-opentelemetry-sdk-bd337d796759) about replacing Jaeger client with [OpenTelemetry](https://opentelemetry.io) SDK (officially) and thought it would be an interesting topic to discuss for a future plan in this package.

I think with the support of OpenTelemetry SDK we'll be able to open up to a wider range of tracing tools. However, the [PHP implementation](https://opentelemetry.io/docs/php/) is still in pre-alpha stage so we're far from doing this anytime soon but I thought of opening up the topic for vigilance.

Here's the [list of vendors](https://opentelemetry.io/vendors/) that support it so far.

Thoughts?

**Comment by adiachenko** [comment]
In premise, I very much agree with the idea of switching to more flexible tooling that won't leave us tied to Jaeger. It would be great to have something that seamlessly integrates with managed offerings like AWS X-Ray and others.

However, the fact that this spawned from OpenCensus makes me slightly skeptical 'cause I tied it shortly before writing this package and I distinctly remember having a bad experience with their instrumentation (it all had a very half-baked and poorly documented feel to it). At the rate this is going, it may take years for this to reach the state where PHP peasants can make a sense of it. 😅

Let's wait and see, I guess. It's also possible that once instrumentation matures we can just use OpenTelemtry directly and this package will transition to being a simple Laravel bridge with a couple of event handlers and middleware.
